src/controllers/poll.js

```diff
@@ -33,15 +34,44 @@ export const getPollsByCategory = async (req, res) => {
     return res.status(200).json({ polls });
   } catch (error) {
     res.status(404).json({ message: error.message });
+    next(error);
   }
 };
 
-export const getPoll = async (req, res) => {
+export const getPoll = async (req, res, next) => {
   try {
     const { id } = req.params;
     const poll = await Poll.findById(id);
     return res.status(200).json({ poll });
   } catch (error) {
     res.status(404).json({ message: error.message });
+    next(error);
+  }
+};
+
+export const makeVote = async (req, res, next) => {
+  try {
+    const { id } = req.params;
+    const { option } = req.body;
+    const poll = await Poll.findById(id);
```

Comment:
could this be `const {results} = await Poll.findById(id);` on one line instead?

Reply:
will do